### PR TITLE
Add utility member functions for partitioned_vector

### PR DESCRIPTION
**Pull request by atrantan**
This pull request does 2 small changes:
- Give a public access to the type named data_type in hpx::server::partitioned_vector
- Add a member action (get_copied_data) allowing to asynchronously get a copy of the data 
  owned by the server

### CODE REVIEW

File: hpx/components/containers/partitioned_vector/partitioned_vector.hpp [hunk 1]
```diff
@@ -732,6 +732,18 @@ namespace hpx
         }
 
     public:
+
+        std::vector< hpx::id_type > get_partitions_ids()
+        {
+            std::vector< hpx::id_type > ids;
+
+            for(auto const part_data : partitions_)
```
Comment (hkaiser) [inline hpx/components/containers/partitioned_vector/partitioned_vector.hpp]: This should probably be: `for(auto const& part_data : partitions_)`

Comment (atrantan) [inline hpx/components/containers/partitioned_vector/partitioned_vector.hpp]: Done in commit 239ed21403b0bb93f56d6a518cff86083ffafde8

Comment (hkaiser) [inline hpx/components/containers/partitioned_vector/partitioned_vector.hpp]: Thanks. Just out of curiosity: why do you need this functionality?

Comment (atrantan) [inline hpx/components/containers/partitioned_vector/partitioned_vector.hpp]: It allows my array_view to get all it needs to return the data of a partition. Saving the ids is costless as well.

Comment (hkaiser) [inline hpx/components/containers/partitioned_vector/partitioned_vector.hpp]: Hmmm, the idea was for the `array_view` to access the local data only. Why do you need to get a copy of the local data for this? Wouldn't it be no _view_ anymore?

Comment (hkaiser) [inline hpx/components/containers/partitioned_vector/partitioned_vector.hpp]: Sorry, misunderstanding - what I wanted to ask is why you need the `get_copied_data()` functionality above.

Comment (atrantan) [inline hpx/components/containers/partitioned_vector/partitioned_vector.hpp]: This is for another functionality extending the concept of array_view. If an array_view needs a data that is not in its locality, make a copy and return it by value.

**Comment by hkaiser** [comment]
> This is for another functionality extending the concept of array_view. If an owner of a array_view needs a data that is not in its locality, make a copy and return it by value. 

Well, let's talk about this in more detail. I'm not convinced yet.

Wrt `get_partitions_ids()`: I don't think you need to add this as all you need for the local access is to create the proper iterators. For instance:

    typedef typename hpx::partitioned_vector<int>::iterator iterator;
    typedef hpx::traits::segmented_iterator_traits<iterator> traits;

    hpx::id_type here = hpx::find_here();
    hpx::partitioned_vector<int> v(...);

    // extract iterators representing local data segments
    auto segment_end = v.segment_end(here);
    for (auto seg_it = v.segment_begin(here); seg_it != segment_end; ++seg_it)
    {
        // extract iterators allowing to (natively) access local data
        auto local_it = traits::begin(seg_it);
        auto local_end = traits::end(seg_it);

        // [local_it, local_end) represents the local data, the iterators
        // are essentially equivalent to the iterators exposed by the 
        // underlying data
    }

So, you can use the exposed iterators directly without even accessing the partition table inside the `partitioned_vector`.

**Pull request merged**